sdk/azidentity/chained_token_credential.go

```diff
@@ -14,15 +14,20 @@ import (
 
 // ChainedTokenCredentialOptions contains optional parameters for ChainedTokenCredential.
 type ChainedTokenCredentialOptions struct {
-	// placeholder for future options
+	// If true, it will not assume the first successful credential should be always used.
+	RetrySources bool
 }
 
 // ChainedTokenCredential is a chain of credentials that enables fallback behavior when a credential can't authenticate.
 type ChainedTokenCredential struct {
-	sources []azcore.TokenCredential
+	sources              []azcore.TokenCredential
+	successfulCredential azcore.TokenCredential
+	retrySources         bool
 }
 
 // NewChainedTokenCredential creates a ChainedTokenCredential.
+// By default, this credential will assume that the first successful credential should be the only credential used on future requests.
```

Comment:
This would be better on the type declaration (rendered docs are [here](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#ChainedTokenCredential)).

Reply:
I moved it here: https://github.com/Azure/azure-sdk-for-go/pull/16392/commits/e2d1201e28ba824b2b640a5eda0b1be1c5120fbc let me know how it looks!